mysite/store/repos/cart_product_repo.py

```diff
@@ -0,0 +1,50 @@
+import json
+import logging
+
+from rest_framework.views import exception_handler
+
+from ..models import CartProduct
+
+logger = logging.getLogger('store')
+
+
+class CartProductRepo:
+    @staticmethod
+    def get_cart_products(cart_pk=None, pk=None):
+        try:
+            if not pk:
+                cart_products = CartProduct.objects.filter(cart_id=cart_pk).select_related('cart','product')
+            else:
+                cart_products = CartProduct.objects.filter(cart_id=cart_pk, product=pk).select_related('cart', 'product')
+            return cart_products
+        except Exception as e:
+            logger.error(f"Error retrieving cart: {e}")
+            return exception_handler(e, "Cart product")
+
+    @staticmethod
+    def create_cart_product(cart_product,product_id,cart_pk):
+        try:
+            cart_product.is_valid()
+            return cart_product.save(product_id=product_id,cart_id=cart_pk)
+        except Exception as e:
+            logger.error(f"Error creating cart product: {e}")
+            return exception_handler(e, "Cart product")
+
+    @staticmethod
+    def delete_cart_product(cart_pk=None, pk=None):
+        try:
+            cart_product = CartProduct.objects.filter(cart_id=cart_pk, product=pk).first()
+            return cart_product.delete()
```

Comment:
Here you deleted the cart-product table, but what if the cart is empty, should we delete it?